### PR TITLE
CONTRIBUTING: Added instructions for PR workflow

### LINKED ISSUE #1104: Can not install development version from source
After installing and activating the virtual environment: 

```
python -m venv ~/virtualenvs/attrsdev/
source ~/.virtualenvs/attrsdev/bin/activate
```
I try to install the development version, but run into the following error: 

```
$ cd repos/attrs/
$ python -m pip install --upgrade pip wheel
$ python -m pip install -e '.[dev]'
.....
ERROR: Cannot install attrs, attrs[dev]==0.1.dev1265, attrs[docs,tests]==0.1.dev1265 and attrs[tests-no-zope]==0.1.dev1265 because these package versions have conflicting dependencies.

The conflict is caused by:
    attrs[dev] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[docs,tests] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[tests-no-zope] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    pytest 7.2.1 depends on attrs>=19.2.0
    attrs[dev] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[docs,tests] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[tests-no-zope] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    pytest 7.2.0 depends on attrs>=19.2.0
    attrs[dev] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[docs,tests] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[tests-no-zope] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    pytest 7.1.3 depends on attrs>=19.2.0
    attrs[dev] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[docs,tests] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[tests-no-zope] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    pytest 7.1.2 depends on attrs>=19.2.0
    attrs[dev] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[docs,tests] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[tests-no-zope] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    pytest 7.1.1 depends on attrs>=19.2.0
    attrs[dev] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[docs,tests] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[tests-no-zope] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    pytest 7.1.0 depends on attrs>=19.2.0
    attrs[dev] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[docs,tests] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[tests-no-zope] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    pytest 7.0.1 depends on attrs>=19.2.0
    attrs[dev] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[docs,tests] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[tests-no-zope] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    pytest 7.0.0 depends on attrs>=19.2.0
    attrs[dev] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[docs,tests] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[tests-no-zope] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    pytest 6.2.5 depends on attrs>=19.2.0
    attrs[dev] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[docs,tests] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[tests-no-zope] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    pytest 6.2.4 depends on attrs>=19.2.0
    attrs[dev] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[docs,tests] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[tests-no-zope] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    pytest 6.2.3 depends on attrs>=19.2.0
    attrs[dev] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[docs,tests] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[tests-no-zope] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    pytest 6.2.2 depends on attrs>=19.2.0
    attrs[dev] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[docs,tests] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[tests-no-zope] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    pytest 6.2.1 depends on attrs>=19.2.0
    attrs[dev] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[docs,tests] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[tests-no-zope] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    pytest 6.2.0 depends on attrs>=19.2.0
    attrs[dev] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[docs,tests] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[tests-no-zope] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    pytest 6.1.2 depends on attrs>=17.4.0
    attrs[dev] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[docs,tests] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[tests-no-zope] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    pytest 6.1.1 depends on attrs>=17.4.0
    attrs[dev] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[docs,tests] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[tests-no-zope] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    pytest 6.1.0 depends on attrs>=17.4.0
    attrs[dev] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[docs,tests] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    attrs[tests-no-zope] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
    pytest 6.0.2 depends on attrs>=17.4.0
    attrs[dev] 0.1.dev1265 depends on attrs 0.1.dev1265 (from /home/user/repos/attrs)
ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```

### PR DESCRIPTION
**Pull request by chrysle**
# Summary

This closes #1104 

# Pull Request Check List

- [ ] Added **tests** for changed code.
  Our CI fails if coverage is not 100%.
- [ ] New features have been added to our [Hypothesis testing strategy](https://github.com/python-attrs/attrs/blob/main/tests/strategies.py).
- [ ] Changes or additions to public APIs are reflected in our type stubs (files ending in ``.pyi``).
    - [ ] ...and used in the stub test file `tests/typing_example.py`.
    - [ ] If they've been added to `attr/__init__.pyi`, they've *also* been re-imported in `attrs/__init__.pyi`.
- [ ] Updated **documentation** for changed code.
    - [ ] New functions/classes have to be added to `docs/api.rst` by hand.
    - [ ] Changes to the signature of `@attr.s()` have to be added by hand too.
    - [ ] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/markup/para.html#directive-versionadded).
          The next version is the second number in the current release + 1.
          So if the current version on PyPI is 23.1.0, the next version is gonna be 23.2.0.
- [ ] Documentation in `.rst` files is written using [semantic newlines](https://rhodesmill.org/brandon/2012/one-sentence-per-line/).
- [x] Changes (and possible deprecations) have news fragments in [`changelog.d`](https://github.com/python-attrs/attrs/blob/main/changelog.d).
- [x] Consider granting [push permissions to the PR branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork), so maintainers can fix minor issues themselves without pestering you.

### CODE REVIEW

File: .github/CONTRIBUTING.md
Comment (webknjaz) [inline .github/CONTRIBUTING.md:110]: Note that `git pull` is a combination of `git fetch` and `git merge`. This may confuse people who are not proficient with Git. I'd go for recommending to always run `git fetch --all` before starting to work on a new PR. I think this would result in less confusion and advanced users would do whatever they want to anyway.

File: .github/CONTRIBUTING.md
Comment (webknjaz) [inline .github/CONTRIBUTING.md:74]: Does GitHub MD renderer always think such references properly? It might be a good idea to use full link syntax here.

File: .github/CONTRIBUTING.md
Comment (webknjaz) [inline .github/CONTRIBUTING.md:105]: The `console` lexer is designed to highlight commands with leading prompt (keeping the optional output uncolored).

```suggestion
$ git checkout -b my_topical_branch upstream/main
```

File: changelog.d/1105.change.md
Comment (webknjaz) [inline changelog.d/1105.change.md:1]: Not sure if the change note is useful for this patch but if it is, I'd probably mark the filename as inline code.
```suggestion
Added instructions for pull request workflow to `CHANGELOG.md`.
```

Comment (chrysle) [reply changelog.d/1105.change.md]: > Not sure if the change note is useful for this patch 

How about adding a [custom fragment](https://towncrier.readthedocs.io/en/stable/configuration.html#custom-fragment-types) like `docs` for towncrier?

**Comment by chrysle** [comment]
@webknjaz I have addressed your comments.

File: changelog.d/1105.change.md
Comment (webknjaz) [reply changelog.d/1105.change.md]: Changelog represents changes between versions, it should show how those changes effect the end-users.
This patch does not affect the end-users at all, only contributors. Honestly, the docs fragment type is rather debatable by itself — I think it's rare for it to be useful.

Comment (chrysle) [reply changelog.d/1105.change.md]: Ok, thanks for your feedback!

**Comment by webknjaz** [comment]
I'll let Hynek decide if it's good enough and merge.

**Comment by chrysle** [comment]
> I'll let Hynek decide if it's good enough and merge.

CC @hynek ;-)

**Comment by hynek** [comment]
5A96CD77-1654-4BA2-9C25-BEB37B60C618

SO busy rn

**Comment by webknjaz** [comment]
Have fun upside-down 🙃

**Comment by hynek** [comment]
As I've already mentioned on the bug report, I'm not super happy to accumulate all git knowledge in my contributing guides which I already find a bit too long (at least too long for everyone to read it ;)).

Does anything speak against just linking to the official docs at https://docs.github.com/en/get-started/quickstart/contributing-to-projects? Maybe with some addendums that are attrs-specific?

**Comment by chrysle** [comment]
> Does anything speak against just linking to the official docs at https://docs.github.com/en/get-started/quickstart/contributing-to-projects? Maybe with some addendums that are attrs-specific?

No, I don't think so. As for the addendums, I'd suggest keeping the `git fetch` commands to highlight the importance of fetching the tags and keeping the local copy up to date.

**Comment by webknjaz** [comment]
I agree with @chrysle on this. In particular, that GitHub docs are incomplete and don't really describe the full workflow.

How about making some bits collapsed via `<details>` so only those needing it would expand and read?

**Comment by webknjaz** [comment]
It's useful to synchronize the remote refs without applying them with merge. `git pull` is the same as `git fetch` + `git merge`. You can use it when you're working on a non-shared branch. Sometimes, it's useful to fetch if you don't intend to merge but want to have remote refs cached/updated locally so you could compare or cherry-pick things.
Naturally, if you work on a single computer and work with your own repository, your changes originate in controlled environments.
But imagine you added other forks as remotes or contribute to another project. In this case, you'd want to update the local copy of those  things that you don't even have write access to. You may then want to rebase on top of upstream's main branch. But `git rebase` does not do network I/O — you specify refs that are already fetched into your `.git/` folder. If you happen to notice that a remote's `main` got new commits, you cannot incorporate those into your branch that you typically push to your fork, unless you first fetch from that remote.
Since I often have several remotes added, I go for `git fetch --all` to synchronize all of them.
Note that things like `git pull` only fetch from one remote, that has the tracking branch, others remain unsynchronized.

Note on `git pull --rebase --autostash` — I use it a lot, mostly for the default branch, but every once in a while, it's easier to just have a preview compared to what's fetched rather than dealing with half-applied stash with conflicts. By the way, using `git worktree` greatly reduces the need to stash stuff.

**Comment by chrysle** [comment]
@webknjaz Thanks for the explanation! It does seem like `git fetch` is quite useful for various development situations and should be kept. 

I'm +1 on your idea with hiding the details. Maybe from [102-109](https://github.com/python-attrs/attrs/pull/1105/files#diff-98e64bc1cd2db9333c6effe87bbe0d6dfe8714aba4c6bde45aa037fe0796e44cR102-R109) and [47-67](https://github.com/python-attrs/attrs/pull/1105/files#diff-98e64bc1cd2db9333c6effe87bbe0d6dfe8714aba4c6bde45aa037fe0796e44cR47-R67)?

File: .github/CONTRIBUTING.md
Comment (webknjaz) [inline .github/CONTRIBUTING.md:46]:
```suggestion
First [fork](https://github.com/python-attrs/attrs/fork) the repository on GitHub.
It will only get a copy of the `main` branch unless you untick the following checkbox:
> * [x] Copy the `main` branch only
```

File: .github/CONTRIBUTING.md
Comment (webknjaz) [inline .github/CONTRIBUTING.md:63]: Since this remote would be read-only to most of the contributors, it's probably better to go for HTTPS in case they don't have SSH set up. If they do have SSH, though, this bit will change nothing and will still work. But I think, this recommendation is more universal.
```suggestion
$ git remote add upstream https://github.com/python-attrs/attrs.git
```

File: .github/CONTRIBUTING.md
Comment (webknjaz) [inline .github/CONTRIBUTING.md:66]: This gives people many options but if they are not proficient with Git, it'll probably only confuse them. OTOH, if they do know Git well, they can come up with this idea on their own. Besides, there's nothing in this guide referring to a remote called `fork` so I'd probably just drop this hint.

File: .github/CONTRIBUTING.md
Comment (webknjaz) [inline .github/CONTRIBUTING.md:74]: This is a bit inaccurate, though. It fetches both branches and tags. But not necessarily all.
Maybe, it makes sense to add `--tags --prune-tags`. Maybe, it makes sense to specify the branch name too.
`git fetch upstream main --tags --prune-tags` would probably represent what we want here.

File: .github/CONTRIBUTING.md
Comment (webknjaz) [reply .github/CONTRIBUTING.md:63]: I've checked out some manpages and it looks like we can also do this:
```suggestion
$ git remote add -t main -m main --tags upstream https://github.com/python-attrs/attrs.git
```

This one-time command will set up `.git/config` to only track the `main` branch and the tags from the `upstream` remote. This means that `git fetch upstream` without args will fetch exactly what we need w/o us having to bother specifying what exactly to fetch.

File: .github/CONTRIBUTING.md
Comment (webknjaz) [reply .github/CONTRIBUTING.md:74]: With the comment in another thread above, it could probably be just `git fetch upstream --prune-tags`.

File: .github/CONTRIBUTING.md
Comment (webknjaz) [inline .github/CONTRIBUTING.md:108]: This could be accompanied with the following snippet:
```console
$ git push -u origin feature/my-topic-branch
```
`-u` marks the branch on the `origin` remote (aka fork) as corresponding to the local one so that `git push` w/o args would pick it up automatically.

File: .github/CONTRIBUTING.md
Comment (webknjaz) [inline .github/CONTRIBUTING.md:110]: This is a bit inaccurate. The local repository does not absorb the tree from upstream. Fetch only updates a local copy of the remotes, separately from the local Git repository tree. We could also tell people that they can sync up their remote fork after fetching with something like `git push origin upstream/main:main` but that's probably an overkill.

**Comment by chrysle** [comment]
Thanks for looking into this! I like your suggestions and will rework my PR if @hynek agrees to.

File: .github/CONTRIBUTING.md
Comment (hynek) [inline .github/CONTRIBUTING.md:66]:
```suggestion
```

File: .github/CONTRIBUTING.md
Comment (hynek) [inline .github/CONTRIBUTING.md:71]:
```suggestion
$ git fetch upstream --prune-tags
```

File: .github/CONTRIBUTING.md
Comment (hynek) [reply .github/CONTRIBUTING.md:105]: I think I prefer a `git fetch --all; git checkout` over this monstrosity.

Traditionally it hasn't been a problem if someone didn't have an up-to-date repo since there's the sweet update branch button. I'd really like to keep this as simple as possible.

Comment (hynek) [inline .github/CONTRIBUTING.md:105]:
```suggestion
```console
$ git fetch --all
$ git checkout -b my_topical_branch upstream/main
```

File: .github/CONTRIBUTING.md
Comment (hynek) [inline .github/CONTRIBUTING.md:113]:
```suggestion
$ git fetch --all
$ git checkout main
$ git merge
```

File: .github/CONTRIBUTING.md
Comment (hynek) [inline .github/CONTRIBUTING.md:108]:
```suggestion
Make your changes, push them to your fork (the remote *origin*):
```console
$ git push -u origin
```

and publish the PR in GitHub's web interface!
```

Comment (hynek) [reply .github/CONTRIBUTING.md:108]: i removed the branch name because it feels tedious and it's unclear to me what problem it's supposed to solve. happy to be corrected, but I'll try to keep it as simple as possible.

Comment (hynek) [reply .github/CONTRIBUTING.md:108]: hrhm looks like GitHub gets confused by the trip backticks

File: .github/CONTRIBUTING.md
Comment (webknjaz) [reply .github/CONTRIBUTING.md:105]: Sure, this example takes things to extremes. Though, it's still a good demonstration because it's accurate.

File: .github/CONTRIBUTING.md
Comment (webknjaz) [reply .github/CONTRIBUTING.md:119]: Which `main` is this pointing at? I'll depend on whether the repo has their local branch in sync with upstream or fork, or is outdated.

File: .github/CONTRIBUTING.md
Comment (chrysle) [reply .github/CONTRIBUTING.md]: This will remove any local tags that do no longer exist in the remote; I guess it's useful.
I think the `--prune` option needs to be passed too for this to succeed? See [git fetch documentation](https://git-scm.com/docs/git-fetch).

Comment (hynek) [reply .github/CONTRIBUTING.md]: yeah that seems too specific for a "get up and contribute real quick 101" for a project that never had to remove a tag. I really want to be helpful, but if we make it too complicated, people won't read / understand it. It's a tedious balance act. :-/

**Comment by hynek** [comment]
@webknjaz 

> I usually let my mentees go through a collection of resources on Git/contributing/reviews: https://gist.github.com/webknjaz/a5a4fb374b7579de827e6bedb93a5220 / https://gist.github.com/webknjaz/cb7d7bf62c3dda4b1342d639d0e78d79 / https://gist.github.com/webknjaz/a7362787a80067af8621a85a71746ca1 / https://gist.github.com/webknjaz/2aa9c7a43b51c1385260ff87e0cbb9e3. There's a lot of things one could learn to be efficient with Git but overwhelming the contributors with such amount of information would be unhelpful (some would even say cruel).

Any chance we could make these resources something more permanent, link to them, and simplify what's in here? Like, even I can't follow all instructions and it's just supposed to get people started to quickly contribute. 🤔

File: .github/CONTRIBUTING.md
Comment (webknjaz) [inline .github/CONTRIBUTING.md:111]: FYI What a push without a branch name does depends on the Git configuration and the defaults depend on the Git version. Before Git 2.0, it'd push all the local branches by default, not just the current one: https://stackoverflow.com/a/948397.
Though, it's probably safe to keep this instruction as is.

`-u` creates a "tracking branch" connection between the local branch and said remote, saving this information in the repo-local Git config.

File: .github/CONTRIBUTING.md
Comment (webknjaz) [reply .github/CONTRIBUTING.md:63]: @hynek I strongly recommend accepting the suggestion in the last comment since it'd make sure that unwanted non-`main` branches won't be fetched by the contributors, creating confusion.

File: .github/CONTRIBUTING.md
Comment (webknjaz) [reply .github/CONTRIBUTING.md:69]: IIRC from reading the manpage two weeks ago, https://github.com/python-attrs/attrs/pull/1105/files#r1146115629 would make this redundant.

Also, `--prune-tags` kinda sets the expectation that you might be removing/force-pushing the upstream tags which is a bad tone. It's equivalent of force-pushing the `main` branch, we just don't do this for a good reason.

That said, you could apply this patch but it shouldn't be necessary in normal situation. Also, according to this guide, this is the first fetch of the upstream after adding it. So at this point, there are no tags in the repository fetched from upstream, hence there's nothing to prune from the local repo anyway.

File: .github/CONTRIBUTING.md
Comment (webknjaz) [reply .github/CONTRIBUTING.md:119]: Not exactly. The connection between a local branch and a remote tracking branch lives in `.git/config`. If it's changed, it may end up merging a remote branch from a different remote than what you might expect. Though, if one were to follow this guide from the beginning (and not change anything in their Git config), I'm almost sure it won't break the assumptions... I actually don't remember ever using `git merge` without the args.

OTOH, assuming that a contributor does not touch their local `main` branch that is configured with `upstream/main` as tracking, I'd replace these three commands with a `git pull --rebase` or `git checkout main && git fetch --all && git reset upstream/main` if the goal is to make the local `main` point to the same commit as the upstream one.

Yet, honestly, I'd only leave the `git fetch --all` command that would sync the remote tree cache and not worry about telling people how to change their local `main`. It's not necessary to do this for making new contributions, given that they follow the instructions of creating their topic branches on top of `upstream/main` and rebasing on top of it, if the need be.

**Comment by webknjaz** [comment]
> Any chance we could make these resources something more permanent, link to them, and simplify what's in here? Like, even I can't follow all instructions and it's just supposed to get people started to quickly contribute.

These gists contain a lot of in-depth materials that are also sometimes redundant. I use them to collect links for my mentees mostly so it's best not to rely on what's inside by linking in the contributing doc but you could extract interesting things to store separately that you could link or include. I mostly posted them because it seems like you or @chrysle might want to lurk inside at some point.

File: .github/CONTRIBUTING.md
Comment (chrysle) [reply .github/CONTRIBUTING.md:69]: And in order to keep everything as simple as possible, options should be rationed ;-). So let's keep this as-is.

File: .github/CONTRIBUTING.md
Comment (chrysle) [reply .github/CONTRIBUTING.md:119]: > Yet, honestly, I'd only leave the `git fetch --all` command that would sync the remote tree cache and not worry about telling people how to change their local `main`. It's not necessary to do this for making new contributions, given that they follow the instructions of creating their topic branches on top of `upstream/main` and rebasing on top of it, if the need be.

I also think this is the most efficient & simple solution.

File: .github/CONTRIBUTING.md
Comment (hynek) [inline .github/CONTRIBUTING.md:123]:
```suggestion
$ git fetch --all
$ git checkout main
$ git merge
```

**Review by hynek — APPROVED** [review]

**Comment by chrysle** [comment]
Yes, this was really getting unoverseeable. 

> pls absolutely feel free to open another PR, but this shed is a McMansion 😅

Will try to apply some leftover feedback and hope you haven't lost your good humour ;-)!

**Pull request merged**
